lasting_effect.cpp

```diff
@@ -752,22 +752,27 @@ bool LastingEffects::canConsume(LastingEffect effect) {
   }
 }
 
-optional<FXName> LastingEffects::getFXName(LastingEffect effect) {
+optional<FXVariantName> LastingEffects::getFX(LastingEffect effect) {
```

Comment:
It could be easier to just return FXDef here instead of the temporary FXVariantName, what do you think?

Reply:
In this case not really, because we would have to store those FXDefs in Creature::particleEffects: it would require a vector<FXDefs> instead of a simple bitset. Also passing an enum is faster than a structure.
It may seem like a micro optimization, but inefficiencies like these tend to accumulate over time and make a whole program much slower unnecessarily.